packages/account/src/providers/utils/extract-tx-error.ts

```diff
@@ -0,0 +1,101 @@
+import { ErrorCode, FuelError } from '@fuel-ts/errors';
+import { bn } from '@fuel-ts/math';
+import type { ReceiptRevert } from '@fuel-ts/transactions';
+import { ReceiptType } from '@fuel-ts/transactions';
+import {
+  FAILED_REQUIRE_SIGNAL,
+  FAILED_ASSERT_EQ_SIGNAL,
+  FAILED_ASSERT_NE_SIGNAL,
+  FAILED_ASSERT_SIGNAL,
+  FAILED_TRANSFER_TO_ADDRESS_SIGNAL,
+  PANIC_REASONS,
+  PANIC_DOC_URL,
+} from '@fuel-ts/transactions/configs';
+
+import type { GqlTransactionStatusFragmentFragment } from '../__generated__/operations';
+import type { TransactionResultReceipt } from '../transaction-response';
+import type { FailureStatus } from '../transaction-summary';
+
+export const assemblePanicError = (status: FailureStatus) => {
+  let errorMessage = `The transaction reverted with reason: "${status.reason}".`;
+
+  if (PANIC_REASONS.includes(status.reason)) {
+    errorMessage = `${errorMessage}\n\nYou can read more about this error at:\n\n${PANIC_DOC_URL}#variant.${status.reason}`;
+  }
+
+  return errorMessage;
+};
+const stringify = (obj: unknown) => JSON.stringify(obj, null, 2);
+
+export const assembleRevertError = (
+  receipts: Array<TransactionResultReceipt>,
+  logs: Array<unknown>
+) => {
+  let errorMessage = 'The transaction reverted with an unknown reason.';
+
+  const revertReceipt = receipts.find(({ type }) => type === ReceiptType.Revert) as ReceiptRevert;
+
+  if (revertReceipt) {
+    const reasonHex = bn(revertReceipt.val).toHex();
+
+    switch (reasonHex) {
+      case FAILED_REQUIRE_SIGNAL: {
+        errorMessage = `The transaction reverted because of a "require" statement has thrown ${
+          logs.length ? stringify(logs[0]) : 'an error.'
+        }.`;
+        break;
+      }
+
+      case FAILED_ASSERT_EQ_SIGNAL: {
+        const sufix =
+          logs.length >= 2 ? ` comparing ${stringify(logs[1])} and ${stringify(logs[0])}.` : '.';
+
+        errorMessage = `The transaction reverted because of an "assert_eq" statement${sufix}`;
+        break;
+      }
+
+      case FAILED_ASSERT_NE_SIGNAL: {
+        const sufix =
+          logs.length >= 2 ? ` comparing ${stringify(logs[1])} and ${stringify(logs[0])}.` : '.';
+
+        errorMessage = `The transaction reverted because of an "assert_ne" statement${sufix}`;
+        break;
+      }
+
+      case FAILED_ASSERT_SIGNAL:
+        errorMessage = `The transaction reverted because of an "assert" statement failed to evaluate to true.`;
```

Comment:
```suggestion
        errorMessage = `The transaction reverted because an "assert" statement failed to evaluate to true.`;
```
This'll break some tests if merged directly as a suggestion.